service/src/main/java/bio/terra/user/app/controller/ProfileApiController.java

```diff
@@ -0,0 +1,56 @@
+package bio.terra.user.app.controller;
+
+import bio.terra.common.iam.SamUser;
+import bio.terra.user.api.ProfileApi;
+import bio.terra.user.model.AnyObject;
+import bio.terra.user.service.user.ProfileService;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+@Controller
+public class ProfileApiController implements ProfileApi {
+
+  private final HttpServletRequest servletRequest;
+  private final ProfileService profileService;
+  private final ProfileSamUserFactory userFactory;
+
+  @Autowired
+  public ProfileApiController(
+      HttpServletRequest servletRequest,
+      ProfileService profileService,
+      ProfileSamUserFactory userFactory) {
+    this.servletRequest = servletRequest;
+    this.profileService = profileService;
+    this.userFactory = userFactory;
+  }
+
+  private SamUser getUser() {
+    return userFactory.from(servletRequest);
+  }
+
+  private List<String> parsePath(String path) {
+    if (path == null || path.length() == 0) return List.of();
+
+    return Arrays.asList(path.split("\\."));
+  }
+
+  @Override
+  public ResponseEntity<Void> userProfileSet(AnyObject body, String path) {
+    profileService.setProperty(getUser(), parsePath(path), body.getValue());
```

Comment:
should we throw 401  if parsePath returns empty array?

Reply:
Agree, since destroying the root seems a bit dangerous. The way the query is set up makes it a no-op if you try that anyway.